File: backend/app/logo-simple.py
import os
import requests
from PIL import Image
import io
from dotenv import load_dotenv

load_dotenv()

API_URL = "https://api-inference.huggingface.co/models/artificialguybr/LogoRedmond-LogoLoraForSDXL-V2"
headers = {"Authorization": "Bearer {os.getenv('API_KEY')}"}

def query(payload):
    response = requests.post(API_URL, headers=headers, json=payload)
    return response.content

# Make sure the /images directory exists
images_dir = os.path.join(os.path.dirname(__file__), 'images')
if not os.path.exists(images_dir):
    os.makedirs(images_dir)

image_bytes = query({
    "inputs": "LogoRedmAF, Icons Stunning Hawaii Sunset",
})

# Load the image from bytes
image = Image.open(io.BytesIO(image_bytes))

# Define the path for the image
image_path = os.path.join(images_dir, 'output_image.jpg')

# Save the image
image.save(image_path)

print(f"Image saved to {image_path}")

File: backend/app/logo-simple.js
/* og code: */

/* async function query(data) {
	const response = await fetch(
		"https://api-inference.huggingface.co/models/artificialguybr/LogoRedmond-LogoLoraForSDXL-V2",
		{
			headers: { Authorization: "Bearer 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" },
			method: "POST",
			body: JSON.stringify(data),
		}
	);
	const result = await response.blob();
	return result;
}
query({"inputs": "LogoRedmAF, Icons " + "Astronaut riding a horse"}).then((response) => {
	// Do something with image
}); */

require('dotenv').config(); // Load environment variables from .env file

const fetch = require('node-fetch');
const fs = require('fs');

async function query(data) {
    const response = await fetch(
        process.env.API_URL,
        {
            headers: { Authorization: `Bearer ${process.env.API_TOKEN}` },
            method: "POST",
            body: JSON.stringify(data),
        }
    );
    const result = await response.buffer(); // Use buffer() instead of blob()

    // Save the image to /images folder
    const path = "./images/image.png"; // Specify the path where you want to save the image
    fs.writeFileSync(path, result); // Write the image to the specified path
    
    return path; // Return the path of the saved image
}

query({"inputs": "LogoRedmAF, Icons " + "Fitness app logo, depicting an abstract figure in motion, vibrant colors"}).then((path) => {
    console.log("Image saved at: ", path);
}).catch((error) => {
    console.error("Error saving image: ", error);
});